src/lstm.js

```diff
@@ -97,10 +100,27 @@ export function lstm(input, weight, recurrentWeight, steps, hiddenSize,
     cellState = nextCell;
 
     if (returnSequence) {
-      nextHidden = reshape(nextHidden, [1, numDirections, null, hiddenSize]);
+      // Expand [numDirections, batchSize, hiddenSize] to
+      // [steps, numDirections, batchSize, hiddenSize]
+      nextHidden = reshape(nextHidden, [1, numDirections, batchSize, hiddenSize]);
+      // Concat output sequence along 0 axis (for steps dimension)
       sequence = (sequence ? concat([sequence, nextHidden], 0) : nextHidden);
     }
   }
 
+  if (direction === 'backward') {
+    // Reverse output sequence alog [0] axes (for steps dimension)
+    sequence = reverse(sequence, {axes: [0]});
```

Comment:
The spec emulation code and Chromium TFLite backend (using emulation code) also have this issue. Please help file issues separately.

Reply:
> The spec emulation code and Chromium TFLite backend (using emulation code) also have this issue. Please help file issues separately.

Done by creating https://github.com/webmachinelearning/webnn/issues/801 for spec
And there's an existed Chromium issue https://issues.chromium.org/issues/385398538 to track.